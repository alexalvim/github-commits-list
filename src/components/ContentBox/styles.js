import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';
import Typo from '../../typo';

const ContentWrapper = styled.div`
  background-color: ${Colors.lightestColor};
  border: solid 1px ${Colors.lightGray};
  border-radius: 5px;
  padding: ${Spaces.double}
  width: 100%;

  ${(props) => props.wrapperStyles}
`;

const Title = styled.h2`
  color: ${Colors.darkestColor};
  font-size: ${Typo.large};
`;

const TitleContainer = styled.div`
  align-items: center;
  display:flex;
  margin-bottom: ${Spaces.base};
`;

const RightContentContainer = styled.div`
  margin-left: auto;
`;

export {
  ContentWrapper,
  Title,
  TitleContainer,
  RightContentContainer
}