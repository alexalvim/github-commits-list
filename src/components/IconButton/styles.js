import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';

const Button = styled.button`
  background-color: ${Colors.mainColor};
  color: ${Colors.lightestColor};
  cursor: pointer;
  padding: ${Spaces.half} ${Spaces.base};
`;

export {
  Button
};