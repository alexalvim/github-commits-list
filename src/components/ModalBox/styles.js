import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';

const OverlayWrapper = styled.div`
  align-items: center;
  background-color: ${Colors.overlayColor}
  display: flex;
  height: 100%;
  justify-content: center;
  left: 0;
  position: fixed;
  top: 0;
  width: 100%;
`;

const contentBoxWrapperStyle = `
  display: flex;
  flex-direction: column;
  height: 100%;

  @media (min-width: ${Spaces.mobileBreakpoint}) {
    max-height: 550px;
    max-width: 450px;
  }
`

export {
  OverlayWrapper,
  contentBoxWrapperStyle
}