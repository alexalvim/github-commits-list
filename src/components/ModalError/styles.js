import styled from 'styled-components';
import { IoIosCloseCircle } from 'react-icons/io';

import Colors from '../../colors';
import Spaces from '../../spaces';
import Typo from '../../typo';

const ErrorIcon = styled(IoIosCloseCircle)`
  color: ${Colors.errorRed};
  margin-right: ${Spaces.half};
`;

const TitleWrapper = styled.div`
  display:flex;
  justify-content: center;
`;

const ErrorMessage = styled.p`
  color: ${Colors.darkestColor};
  font-size: ${Typo.medium};
`;

export {
  ErrorIcon,
  TitleWrapper,
  ErrorMessage
};