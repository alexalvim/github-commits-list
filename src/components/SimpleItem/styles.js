import styled from 'styled-components';

import Colors from '../../colors';
import Typo from '../../typo';
import Spaces from '../../spaces';

const Title = styled.h3`
  color: ${Colors.mainColor};
  cursor: pointer;
  font-size: ${Typo.medium};
  font-weight: bold;
  margin-bottom: ${Spaces.half};

  &:hover {
    text-decoration: underline;
  }
`;

const Description = styled.p`
  font-size: ${Typo.medium};
`;

const ContentWrapper = styled.div`
  padding: ${Spaces.double} 0;
`;

export {
  Title,
  Description,
  ContentWrapper
}