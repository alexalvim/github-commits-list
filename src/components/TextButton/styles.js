import styled from 'styled-components';

import Colors from '../../colors';
import Typo from '../../typo';

const Button = styled.button`
  background: none;
  border: none;
  color: ${Colors.mainColor};
  cursor: pointer;
  font-size: ${Typo.medium};

  ${(props) => props.wrapperStyle}
`;

export {
  Button
};