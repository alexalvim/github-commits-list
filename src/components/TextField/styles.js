import styled from 'styled-components';

import Colors from '../../colors';
import Spaces from '../../spaces';
import Typo from '../../typo';

const Input = styled.input`
  background-color: ${Colors.lightestColor};
  color: ${Colors.darkestColor}
  font-size: ${Typo.medium};
  padding: ${Spaces.half};

  ${(props) => props.wrapperStyle}
`;

export {
  Input
};