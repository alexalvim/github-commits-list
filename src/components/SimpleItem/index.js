import React from 'react';

import { ContentWrapper, Title, Description } from './styles';

export default ({ title, description, onClick }) => (
  <ContentWrapper>
    <Title onClick={onClick}>{title}</Title>
    {description && <Description>{description}</Description>}
  </ContentWrapper>
);