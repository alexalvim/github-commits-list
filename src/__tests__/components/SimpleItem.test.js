import React from 'react';
import { shallow } from 'enzyme';

import SimpleItem from '../../components/SimpleItem';
import { Title, Description } from '../../components/SimpleItem/styles';

describe('Testing SimpleItem', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<SimpleItem title='test' description='desc' />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly without description', () => {
    const wrapper = shallow(<SimpleItem title='test' />);

    expect(wrapper.find(Description).length).toEqual(0);
  });

  it('call expected function', () => {
    const mockFunction = jest.fn();
    const wrapper = shallow(<SimpleItem title='test' onClick={mockFunction} />);

    wrapper.find(Title).simulate('click');
    expect(mockFunction).toHaveBeenCalled();    
  })
});