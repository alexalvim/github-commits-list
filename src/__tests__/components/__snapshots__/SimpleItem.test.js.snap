// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SimpleItem should render correctly 1`] = `ShallowWrapper {}`;
